### ISSUE TITLE
After installation on Ubuntu, it cannot be started normally due to a permission problem

### ISSUE THREAD

**Issue by flyratlove** [jgraph/drawio-desktop#1926]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio-desktop/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

1.  On Ubuntu, I downloaded the amd64.deb from GitHub and completed the installation without any issues.
2. Clicking the drawio icon did not start the application.
3. When trying to start drawio by typing it in the terminal, it prompted that the permissions for `/opt/drawio/chrome-sandbox` were incorrect and needed to be set to `4755`.
4. Following the prompt, I executed `sudo chmod 4755 /opt/drawio/chrome-sandbox` in the terminal.
5. Subsequently, drawio started successfully.

From the above steps, it can be confirmed that there was an issue with the permissions configuration of `chrome-sandbox` in the `drawio amd64.deb package`. I am not sure if this issue is specific to `Ubuntu 24.10 `or if it occurs on all Linux distributions. Regardless, after the above operation, drawio is now working properly.

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 25.0.2/26.0.3/26.0.4

**Desktop (please complete the following information):**
 - OS: Ubuntu24.10

**Comment by grf692** [comment]
FYI it also affects KDE neon 6.2 (which based on Ubuntu 24.04).

**Comment by ee5cd7d5d** [comment]
It affects the appimage as well. I am using Ubuntu 24.04, and my error message is:

```
[27832:1216/103416.320669:FATAL:setuid_sandbox_host.cc(163)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /tmp/.mount_drawio8DR1lB/chrome-sandbox is owned by root and has mode 4755.
[1]    27832 trace trap (core dumped)  ./drawio-x86_64-25.0.2.AppImage
[1216/103416.439863:ERROR:icu_util.cc(223)] Invalid file descriptor to ICU data received.
```

**Comment by xcfstudio** [comment]
我使用Ubuntu系统，KDE桌面环境，遇到同样问题。在控制台输入`sudo chmod 4755 /opt/drawio/chrome-sandbox`即可解决问题。

I am using Ubuntu system and KDE desktop environment, and I am experiencing the same problem. To solve the problem, enter `sudo chmod 4755/opt/rawio/chrome sandbox` in the console.

**Comment by krichter722** [comment]
Reoccurring problem since 2019, the downloadable packages and images just don't work.

https://github.com/jgraph/drawio-desktop/issues/158
https://github.com/jgraph/drawio-desktop/issues/150
and many more.

Apparently devs consider it a "fix" if users chmod the directory in question. Therefore the issue arises again and again over many years. I guess for the AppImage which uses a temporary directory there's no fix and it's conceptually unusable.

If you find and understand the message you can work around it with `chmod` , however I think you shouldn't because setuid comes with a ton of implications and most users won't know what they're doing and risking. And it's simply against any common sense of security to leak unnecessary permissions by design.

**Comment by davidjgraph** [comment]
We don't consider it anything, drop the attitude. We just don't run linux boxes and are bored of 90% of OS issues being linux, it's a pile of shit to develop for.

**Comment by lucaaamaral** [comment]
I just wanted to reinforce here that I'm facing the same issue with the .deb installer. It also happens after an update ran through the application.
It is just a small problem on a great application.

**Comment by complib2891** [comment]
seems to me this application is the perfect target for hacker, since the sandbox is own by root, thus when running the desktop app, these security holes are candidates:

XSS since it's javascript based, thus can run every commands on this system when this app is open. so hackers just need to implement supply-chain-attack behind the scene by targeting vulnerable packages, then you can take over developers machines from all over the world, it's ripe for the taking. hohoho~

I hope hackers and security researchers can pitch in on this.. as I'm not and I'm worried. I'm sorry as I'm one of 90% not the 10%

**Comment by davidjgraph** [comment]
So, just to clarify. The constructive solution is to submit a PR. If you behave like a 4 year old, like our friend @complib2891 , you'll just be banned.

**Comment by rmsilva1973** [comment]
@davidjgraph People behaving like a 4 year old should be kicked out indeed. BUT... Doesn't the README.md for the project states that it isn't open to PRs?

**Comment by davidjgraph** [comment]
I'm allowed to bend any rules I like :). I'll take a PR on this, I don't know what to change and don't have an Ubutnu install to start testing.

But please note, there's a trend to create a chatgpt suggestion of PR, submit it and say "ok there". The PR doesn't work and we have to go and work out the whole thing from scratch and debug it ourselves. So, PRs are OK here, but fully tested PRs...

**Comment by Hugo-Trentesaux** [comment]
Seems to be the same cause as #1809

**Comment by brynjare** [comment]
I ran into this earlier today, and started digging around a bit. This error message is likely caused by a security change originally [introduced in AppArmor on Ubuntu 23.10](https://ubuntu.com/blog/ubuntu-23-10-restricted-unprivileged-user-namespaces).

It also affects a bunch of other Electron applications, and from what I can tell the chrome-sandbox fix is based on a [deprecated method](https://github.com/electron-userland/electron-builder/pull/8368). The official solution seems to be to [create an AppArmor profile](https://github.com/electron-userland/electron-builder/issues/8635) instead.

To test this theory, and as a temporary workaround, I created a new AppArmor profile manually on a system running Ubuntu 24.04, with the file name `/etc/apparmor.d/drawio`:
```
# This profile allows everything and only exists to give the
# application a name instead of having the label "unconfined"

# Manually created 2025-02-25 to fix sandbox errors

abi <abi/4.0>,
include <tunables/global>

profile drawio /opt/drawio/drawio flags=(unconfined) {
  userns,

  # Site-specific additions and overrides. See local/README for details.
  include if exists <local/drawio>
}
```
After reloading apparmor (with either `sudo apparmor_parser -r /etc/apparmor.d/drawio` or `sudo systemctl reload apparmor.service`) I'm able to launch drawio without changing the permissions of `/opt/drawio/chrome-sandbox`.

Unfortunately I'm not familiar enough with Electron to try to submit a pull request, but https://github.com/electron-userland/electron-builder/pull/8636 might contain some hints on where to start if anyone is inclined to give it a go.

**Comment by phaze-phusion** [comment]
As previously stated the safe way to solve for the `chrome-sandbox` permissions issue is to use an AppArmor profile.

The method described by [brynjare](https://github.com/brynjare) works if you want to do it manually.

For those who just want to execute a script: Here's one I adapted from a script I used for Firefox beta.

```bash
#!/bin/bash

# Add an AppArmor profile for DrawIO to use chrome-sandbox without permission issues
# adapted for drawio from https://support.mozilla.org/en-US/kb/install-firefox-linux#w_security-features-warning
#
# execute this script with sudo

profileIdentifier="drawio"
installationDirectory="/opt/drawio" # no trailing slash

printf -v fileCreationDate '%(%Y-%m-%d)T' -1

cat << EOF > /etc/apparmor.d/$profileIdentifier
# This profile allows everything and only exists to give the
# application a name instead of having the label 'unconfined'

# Manually created $fileCreationDate to fix chrome-sandbox issue

abi <abi/4.0>,
include <tunables/global>
profile $profileIdentifier
$installationDirectory/$profileIdentifier
flags=(unconfined) {
  userns,

  # Site-specific additions and overrides. See local/README for details.
  include if exists <local/$profileIdentifier>
}
EOF

systemctl restart apparmor.service
```

Edit: I read that AppArmor profile support is coming to the Electron Builder soon. I, like most linux user's will be fine with this manual workaround till all Electron apps can benefit from the new builder.

PS: Thank you to the drawio-desktop maintainers for this wonderful app

**Comment by davidjgraph** [comment]
It does seem to be https://github.com/electron-userland/electron-builder/issues/8635.

**Comment by brynjare** [comment]
Yeah - looks like electron-builder v25.1.8 is still the most recent stable release (based on the `latest` tag), but from what I can tell the issue will resolve itself once the various projects are able to update to electron-builder v26.x.

**Comment by brynjare** [comment]
I just updated to v26.2.2 and can confirm that it works out of the box for me now on Ubuntu 24.04, without any local apparmor modifications.
(I manually removed my local config before updating, and a new config was auto-created during installation.)
